crates/sui-rpc-api/proto/sui/rpc/v2alpha/execution_status.proto

```diff
@@ -0,0 +1,335 @@
+
+    // Type parity mismatch for Move function.
+    // Mismatch between the number of actual versus expected type arguments.
+    EXECUTION_ERROR_KIND_TYPE_ARITY_MISMATCH = 18;
+
+    // Non-entry function invoked. Move Call must start with an entry function.
+    EXECUTION_ERROR_KIND_NON_ENTRY_FUNCTION_INVOKED = 19;
+
+    // Invalid command argument.
+    EXECUTION_ERROR_KIND_COMMAND_ARGUMENT_ERROR = 20;
+
+    // Type argument error.
+    EXECUTION_ERROR_KIND_TYPE_ARGUMENT_ERROR = 21;
+
+    // Unused result without the drop ability.
+    EXECUTION_ERROR_KIND_UNUSED_VALUE_WITHOUT_DROP = 22;
+
+    // Invalid public Move function signature.
+    // Unsupported return type for return value.
+    EXECUTION_ERROR_KIND_INVALID_PUBLIC_FUNCTION_RETURN_TYPE = 23;
+
+    // Invalid transfer object, object does not have public transfer.
+    EXECUTION_ERROR_KIND_INVALID_TRANSFER_OBJECT = 24;
+
+    //
+    // Post-execution errors.
+    //
+
+    // Effects from the transaction are too large.
+    EXECUTION_ERROR_KIND_EFFECTS_TOO_LARGE = 25;
+
+    // Publish or Upgrade is missing dependency.
+    EXECUTION_ERROR_KIND_PUBLISH_UPGRADE_MISSING_DEPENDENCY = 26;
+
+    // Publish or upgrade dependency downgrade.
+    //
+    // Indirect (transitive) dependency of published or upgraded package has been assigned an
+    // on-chain version that is less than the version required by one of the package's
+    // transitive dependencies.
+    EXECUTION_ERROR_KIND_PUBLISH_UPGRADE_DEPENDENCY_DOWNGRADE = 27;
+
+    // Invalid package upgrade.
+    EXECUTION_ERROR_KIND_PACKAGE_UPGRADE_ERROR = 28;
+
+    // Indicates the transaction tried to write objects too large to storage.
+    EXECUTION_ERROR_KIND_WRITTEN_OBJECTS_TOO_LARGE = 29;
+
+    // Certificate is on the deny list.
+    EXECUTION_ERROR_KIND_CERTIFICATE_DENIED = 30;
+
+    // Sui Move bytecode verification timed out.
+    EXECUTION_ERROR_KIND_SUI_MOVE_VERIFICATION_TIMEDOUT = 31;
+
+    // The requested shared object operation is not allowed.
+    EXECUTION_ERROR_KIND_SHARED_OBJECT_OPERATION_NOT_ALLOWED = 32;
+
+    // Requested shared object has been deleted.
+    EXECUTION_ERROR_KIND_INPUT_OBJECT_DELETED = 33;
+
+    // Certificate is canceled due to congestion on shared objects.
+    EXECUTION_ERROR_KIND_EXECUTION_CANCELED_DUE_TO_SHARED_OBJECT_CONGESTION = 34;
+
+    // Address is denied for this coin type.
+    EXECUTION_ERROR_KIND_ADDRESS_DENIED_FOR_COIN = 35;
+
+    // Coin type is globally paused for use.
+    EXECUTION_ERROR_KIND_COIN_TYPE_GLOBAL_PAUSE = 36;
+
+    // Certificate is canceled because randomness could not be generated this epoch.
+    EXECUTION_ERROR_KIND_EXECUTION_CANCELED_DUE_TO_RANDOMNESS_UNAVAILABLE = 37;
+  }
+}
+
+// A size error.
+message SizeError {
+  // The offending size.
+  optional uint64 size = 1;
+  // The maximum allowable size.
+  optional uint64 max_size = 2;
+}
+
+// Location in Move bytecode where an error occurred.s
+message MoveLocation {
```

Comment:
Is it worth including source line here in anticipation of clever errors, or would you rather wait?

Reply:
Either way, if we already have an idea of the shape of this info and a way on-node to populate it we can do it now.